src/noInputPrefixRule.ts

```diff
@@ -0,0 +1,46 @@
+import * as Lint from 'tslint';
+import * as ts from 'typescript';
+import { sprintf } from 'sprintf-js';
+import { NgWalker } from './angular/ngWalker';
+
+export class Rule extends Lint.Rules.AbstractRule {
+  public static metadata: Lint.IRuleMetadata = {
+    ruleName: 'no-input-prefix',
+    type: 'maintainability',
+    description: 'Name inputs without the prefix with the provided prefixes',
+    rationale: `HTML attributes are not prefixed. It's condered best not to use 'is', 'should', 'must' prefixes for boolean values inputs.`,
+    options: {
+      'type': 'array',
+      'items': [
+        { 'type': 'string' }
+      ],
+    },
+    optionsDescription: 'Not configurable.',
+    typescriptOnly: true
+  };
+
+  static FAILURE_STRING: string = 'In the class "%s", the input property "%s" should not be prefixed with %s';
+
+  public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+    return this.applyWithWalker(new InputWalker(sourceFile, this.getOptions()));
+  }
+}
+
+class InputWalker extends NgWalker {
+  visitNgInput(property: ts.PropertyDeclaration, input: ts.Decorator, args: string[]) {
+    const className = (<any>property).parent.name.text;
+    const memberName = (<any>property.name).text as string;
+    const options = this.getOptions() as string[];
+    let prefixLength = 0;
+
+    if (
```

Comment:
This rule isn't only for boolean Input ?

Reply:
In reality it can work for all type of inputs

Reply:
Unless we want to limit it to booleans only